File: src/main/java/com/aleprimo/Booking_System_App/service/PaymentService.java
package com.aleprimo.Booking_System_App.service;



import com.aleprimo.Booking_System_App.entity.Payment;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface PaymentService {
    Payment createPayment(Payment payment);
    Optional<Payment> getPaymentById(Long id);
    Optional<Payment> getPaymentByBookingId(Long bookingId);
    Page<Payment> getAllPayments(Pageable pageable);
    void deletePayment(Long id);
}

File: booking-frontend/src/services/paymentService.js
import api from "./api";

export const getPayments = (page = 0, size = 10) =>
  api.get(`/payments?page=${page}&size=${size}`);

export const getPaymentById = (id) => api.get(`/payments/${id}`);

export const createPayment = (data) => api.post("/payments", data);

export const deletePayment = (id) => api.delete(`/payments/${id}`);
